src/Solver/time_stepping/GenericGhostTimeCluster.h

```diff
@@ -6,23 +6,22 @@
 #include "AbstractTimeCluster.h"
 
 namespace seissol::time_stepping {
-
-class GhostTimeCluster : public AbstractTimeCluster {
- private:
+class GenericGhostTimeCluster : public AbstractTimeCluster {
```

Comment:
IMO the design should be different. Both GhostTimeClusters should inherit from AbstractGhostTimeCluster, which defines the interface.
This here should be called DirectGhostTimeCluster or simular.
testQueue can be a non-class function (poor design on my part there!)

Reply:
> Both GhostTimeClusters should inherit from AbstractGhostTimeCluster, which defines the interface.

Done. I moved all common parts to `AbstractGhostTimeCluster`. Both `DirectGhostTimeCluster` and `DirectGhostTimeCluster` add their own behavior i.e., `sendCopyLayer`, `receiveGhostLayer` and `testForGhostLayerReceives`

> testQueue can be a non-class function
I don't know what to do with it. As for me, it is not super critical here.